proposal-templates/0000-swift-template.md

```diff
@@ -47,107 +75,144 @@ reasonably implement the feature.
+for users of the library?  If a library adopts the feature, can it
+be *un*-adopted later withour breaking source or ABI compatibility?
+Will package authors be able to selectively adopt this feature depending
+on the tools version available, or will it require bumping the minimum
+tools version required by the package?
+
+If there are no concerns to raise in this section, it may be removed.
+
+## Future directions
+
+Describe any interesting proposals that could build on this proposal
+in the future.  This is especially important when these future
+directions inform the design of the proposal, for example by making
+sure an attribute encodes enough information to be used for other
+purposes.
+
+The rest of the proposal should generally not talk about future
+directions except by referring to this section.  It is important
+not to confuse reviewers about what is covered by this specific
+proposal.  If there's a larger vision that needs to be explained
+in order to understand this proposal, consider starting a discussion
+thread on the forums to capture your broader thoughts.
+
+Avoid making affirmative statements in this section, such as "we
+will" or even "we should".  Describe the proposals neutrally as
+possibilities to be considered in the future.
+
+Consider whether any of these future directions should really just
+be part of the current proposal.  It's important to make focused,
+self-contained proposals that can be incrementally implemented and
+reviewed, but it's also good when proposals feel "complete" rather
+than leaving significant gaps in their design.  For example, when
+[SE-0193](https://github.com/apple/swift-evolution/blob/main/proposals/0193-cross-module-inlining-and-specialization.md)
+introduced the `@inlinable` attribute, it also included the
+`@usableFromInline` attribute so that declarations used in inlinable
+functions didn't have to be `public`.  This was a relatively small
+addition to the proposal which avoided creating a serious usability
+problem for many adopters of `@inlinable`.
 
 ## Alternatives considered
 
-Describe alternative approaches to addressing the same problem, and
-why you chose this approach instead.
+Describe alternative approaches to addressing the same problem.
+This is an important part of most proposal documents.  Reviewers
+are often familiar with other approaches prior to review and may
+have reasons to prefer them.  This section is your first opportunity
+to try to convince them that your approach is the right one, and
+even if you don't fully succeed, you can help set the terms of the
+conversation and make the review a much more productive exchange
+of ideas.
+
+You should be fair about other proposals, but you do not have to
+be neutral; after all, you are specifically proposing something
+else.  Describe any advantages these alternatives might have, but
+also be sure to explain the disadvantages that led you to prefer
+the approach in this proposal.
```

Comment:
Perhaps also worth a brief passage to the effect of "this section should be updated with any particularly compelling alternatives that are raised while the proposal is in the pitch phase"?

Reply:
Good idea, incorporated.